### DISCUSSION TITLE
Version 2.2

### DISCUSSION THREAD

**Discussion by Jaysmito101** [Jaysmito101/TerraForge3D#25] [Announcements]
# Features

TerraForge3D has got a series of features. Listing all of them is a very difficult task. Here are some of the major features.

### GPU Powered Terrain Generation

TerraForge3D utilizes the power of your GPU to generate the terrains blazing fast. TerraForge3D utilizes OpenCL 2.0 to use your GPU. With GPU mesh generation you can work with extremely huge meshs in realtime.

### CPU Powered Terrain Generation

There are somethings which cannot possible be done on a GPU as efficiently(like the node editor) so you can optionally mix GPU generation with CPU generation to get more control. Also if you GPU doesnt support OpenCL 2.0 you can still use the CPU mesh Generation. Also TerraForge's CPU Mesh Generation algorithms are made keeping efficiency and speed in mind thus they are extremely fast.

### No Limit on Resolution

Most terrain generation applications limit you to a certain maximum resolution or only certain resolutions like (128, 256, ...) but not TerraForge3D! TerraForge3D Allows you to work with any resolution you can imagine of.

### Advanced Node Editor

TerraForge3D has a very advanced Node Editor with 40+ different nodes which let you completely control how the terrain is generated. You can even use mathematical functions to shape your terrain!

### Physically Based Erosion

TerraForge3D has several erosion algorithms(hydraulic, wind, ...) which lets you simulate the simulation on you procedurally generated terrains giving it an extra touch of realism. These algorithms simulate actual rain, wind, dust to give the best effect.

### Custom Shape Generation

Procedural is great! But sometimes there must be a human touch! Thus TerraForge3D lets you customize your terrain by pmanyally placing terrrain features like hills, caters, cliffs, etc and combine it with procedural terrain.

### Custom Sky

TerraForge3D allows you to completely customize the sky. You can use a sky box, sky dome. You can procedurally generate a sky or use a HDRI.

### Sea Level Visualization

TerraForge3D also lets you visualize a sea to give you a idea of the sea level.

### Custom Shaders

TerraForge3D allows you to have a custom shader(in GLSL). TerraForge3D also provides a inbuilt editor for writing your shaders!

### Texture Store & Texture Settings

TerraForge3D gives you a huge collection of textures to download and use free of cost! Also TerraForge3D's texture settings give you fine control over how the mesh is textured.

### Universal

TerraForge3D can export to a whole lot of 3D file formats like OBJ, GLTF, GLB, STL, ... . Also you can export heightmaps as PNG, JPG, ... . This makes TerraForge3D totally compitable with any program you can think of! Be it Blender, Unreal Engine, Uniy, Maya, anything.

<hr /><em>This discussion was created from the release <a href='https://github.com/Jaysmito101/TerraForge3D/releases/tag/v2.2'>Version 2.2</a>.</em>

**Comment by XxnittanixX** [comment]
its got a ui rendering bug  make it bigger

**Comment by Jaysmito101** [comment]
@XxnittanixX dould you explain what are ypuntalkimg of exactly?
